File: frontend/src/types/trip.ts
export interface TripPlanRequest {
  departure_city?: string
  city: string
  start_date: string
  days: number
  budget: number
  people: number
  preference: string
  requirements?: string
}

export interface Location {
  longitude: number
  latitude: number
}

export interface Attraction {
  name: string
  address: string
  location: Location
  visit_duration: number
  ticket_price: number
  description: string
  image_url?: string
  category?: string
}

export interface Hotel {
  name: string
  address: string
  price: number
  description: string
  location?: Location
}

export interface WeatherInfo {
  date: string
  weather: string
  temperature: string
  suggestion: string
}

export interface DayPlan {
  day: number
  title: string
  attractions: Attraction[]
  meals: string[]
  hotel: Hotel
  weather: WeatherInfo
}

export interface Budget {
  total_attractions: number
  total_hotels: number
  total_meals: number
  total_transportation: number
  total: number
}

export interface RequirementSummary {
  raw_text: string
  pace: string
  companions: string[]
  food_preferences: string[]
  hotel_preferences: string[]
  avoid: string[]
  route_preferences: string[]
  attractions_per_day: number
  mobility_level: string
  route_intensity: string
  meal_focus: string
  hotel_area_preference: string
  must_have: string[]
  must_avoid: string[]
}

export interface TripPlan {
  city: string
  start_date: string
  days: DayPlan[]
  budget: Budget
  overall_suggestion: string
  requirement_summary?: RequirementSummary
  transport_summary?: TransportSummary
}

export interface TransportOption {
  mode: string
  title: string
  description: string
  estimated_duration: string
  estimated_cost: string
  booking_advice: string
}

export interface TransportSummary {
  departure_city?: string | null
  destination_city: string
  recommended_mode: string
  summary: string
  options: TransportOption[]
}

export interface SavedTripPlan {
  id: string
  title: string
  city: string
  start_date: string
  days_count: number
  budget_total: number
  saved_at: string
  updated_at: string
  trip_plan: TripPlan
}

File: backend/app/models/trip.py
from typing import List, Optional

from pydantic import BaseModel, Field


class TripPlanRequest(BaseModel):
    city: str = Field(..., description="目的地城市")
    start_date: str = Field(..., description="出发日期")
    days: int = Field(..., ge=1, le=15, description="游玩天数")
    budget: int = Field(..., ge=0, description="总预算")
    people: int = Field(..., ge=1, le=20, description="出行人数")
    preference: str = Field(..., description="旅行偏好")
    requirements: Optional[str] = Field(None, description="补充需求")


class Location(BaseModel):
    longitude: float
    latitude: float


class Attraction(BaseModel):
    name: str
    address: str
    location: Location
    visit_duration: int
    ticket_price: int
    description: str
    image_url: Optional[str] = None
    category: str = "景点"


class Hotel(BaseModel):
    name: str
    address: str
    price: int
    description: str
    location: Optional[Location] = None


class WeatherInfo(BaseModel):
    date: str
    weather: str
    temperature: str
    suggestion: str


class DayPlan(BaseModel):
    day: int
    title: str
    attractions: List[Attraction]
    meals: List[str]
    hotel: Hotel
    weather: WeatherInfo


class Budget(BaseModel):
    total_attractions: int
    total_hotels: int
    total_meals: int
    total_transportation: int
    total: int


class RequirementSummary(BaseModel):
    raw_text: str = ""
    pace: str = "正常"
    companions: List[str] = []
    food_preferences: List[str] = []
    hotel_preferences: List[str] = []
    avoid: List[str] = []
    route_preferences: List[str] = []
    attractions_per_day: int = 2


class TripPlan(BaseModel):
    city: str
    start_date: str
    days: List[DayPlan]
    budget: Budget
    overall_suggestion: str
    requirement_summary: Optional[RequirementSummary] = None
